src/uct/ib/ud/base/ud_ep.c

```diff
@@ -98,12 +98,29 @@ static void uct_ud_ep_reset(uct_ud_ep_t *ep)
     ep->resend.pos       = ucs_queue_iter_begin(&ep->tx.window);
     ep->resend.psn       = ep->tx.psn;
     ep->resend.max_psn   = ep->tx.acked_psn;
+    ep->resend.retry_cnt = 0;
 
     ep->rx.acked_psn = UCT_UD_INITIAL_PSN - 1;
     ucs_frag_list_init(ep->tx.psn-1, &ep->rx.ooo_pkts, 0 /*TODO: ooo support */
                        UCS_STATS_ARG(ep->rx.stats));
 }
 
+static void uct_ud_ep_timed_out(uct_ud_iface_t *iface, uct_ud_ep_t *ep)
+{
+    ucs_error("Resend error on ep=%p: peer is not responding", ep);
```

Comment:
If we go for fault tolerance, we will have to remove these verbose messages later on.